docs/framework/react/guide/custom-link.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Custom Link
+---
+
+While repeating yourself can be fine in many cases, you may find yourself repeating yourself too much. Sometimes you may want to make cross cutting components with extra behavior or styles. Or sometimes you may also use third party libraries and combine it with TanStack Router's type safety
+
+# `createLink` for cross cutting concerns
+
+`createLink` creates a custom `Link` component with the same type parameters as `Link`. This means you can create your own component which provides the same type safety and typescript performance as `Link`.
+
+If you want to create a component which wraps `Link` with some additional styles or behavior then you can do so with the following
+
+```tsx
+import { Link, createLink } from '@tanstack/react-router'
+
+export const NavigationLink = createLink(
+  React.forwardRef((props: {}, ref: React.Ref<HTMLAnchorElement>) => {
+    return (
+      <Link
+        {...props}
+        ref={ref}
+        className="block py-2 px-3 text-blue-700"
+        preload="intent"
+        activeProps={{ className: `font-bold` }}
+      />
+    )
+  }),
+)
+```
+
+> React.forwardRef wil not be required in React 19 -->
+
+You can then use your newly created `Link` component as any other `Link`
+
+```tsx
+<NavigationLink
+  to="/dashboard/invoices/$invoiceId"
+  params={{
+    invoiceId: 0,
+  }}
+/>
+```
+
+# `createLink` with third party libraries
+
+You might want to use third party component libraries with TanStack Router. For example to use `Button` from MUI you can use `createLink` which infers the types from both `Button` and `Link` while keeping type parameters necessary for TanStack Router's type safety
+
+```tsx
+const ButtonLink = createLink(
+  React.forwardRef(
+    (
+      props: Parameters<typeof Button>[0],
+      ref: React.Ref<HTMLButtonElement>,
+    ) => {
+      return <Button {...props} ref={ref} component={Link} />
```

Comment:
and how does this button then get acess to the `to`?

Reply:
Does it need to? Its there at runtime but its not been typed. I'm not sure the use case for this though

Reply:
what if you want to navigate to `to` upon button click?

Reply:
We have a similar usecase for getting access to the `active: boolean` property to render a different variant of the component based on it being active. 

_(`activeProps` doesn't seem to propagate as expected)_